forum/backends/mysql/models.py

```diff
@@ -0,0 +1,576 @@
+                content_object_id=self.pk, content_type=self.content_type
+            ).values_list("user_id", flat=True)
+        )
+
+    @property
+    def edit_history(self) -> QuerySet[EditHistory]:
+        """Return a list of edit history for the content."""
+        return EditHistory.objects.filter(
+            content_object_id=self.pk, content_type=self.content_type
+        )
+
+    @property
+    def votes(self) -> models.QuerySet[UserVote]:
+        return UserVote.objects.filter(
+            content_object_id=self.pk,
+            content_type=self.content_type,
+        )
+
+    class Meta:
+        abstract = True
+
+
+class CommentThread(Content):
+    """Comment thread model."""
+
+    THREAD_TYPE_CHOICES = [
+        ("question", "Question"),
+        ("discussion", "Discussion"),
+    ]
+
+    CONTEXT_CHOICES = [
+        ("course", "Course"),
+        ("standalone", "Standalone"),
+    ]
+
+    title: models.CharField[str, str] = models.CharField(max_length=255)
+    thread_type: models.CharField[str, str] = models.CharField(
+        max_length=50, choices=THREAD_TYPE_CHOICES, default="discussion"
+    )
+    context: models.CharField[str, str] = models.CharField(
+        max_length=50, choices=CONTEXT_CHOICES, default="course"
+    )
+    closed: models.BooleanField[bool, bool] = models.BooleanField(default=False)
+    pinned: models.BooleanField[Optional[bool], bool] = models.BooleanField(
+        null=True, blank=True
+    )
+    comment_count: models.PositiveIntegerField[int, int] = models.PositiveIntegerField(
+        default=0
+    )
+    last_activity_at: models.DateTimeField[Optional[datetime], datetime] = (
+        models.DateTimeField(null=True, blank=True)
+    )
+    close_reason_code: models.CharField[Optional[str], str] = models.CharField(
+        max_length=255, null=True, blank=True
+    )
+    closed_by: models.ForeignKey[User, User] = models.ForeignKey(
+        User,
+        related_name="threads_closed",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+    )
+
+    @classmethod
+    def get(cls, thread_id: str) -> CommentThread:
+        """Get a comment thread model instance."""
+        return cls.objects.get(pk=int(thread_id))
+
+    def to_dict(self) -> dict[str, Any]:
+        """Return a dictionary representation of the model."""
+        votes: dict[str, Any] = {
+            "up": [],
+            "down": [],
+            "up_count": 0,
+            "down_count": 0,
+            "count": 0,
+            "point": 0,
+        }
+        for vote in self.votes.all():
+            if vote.vote == 1:
+                votes["up"].append(vote.user.pk)
+                votes["up_count"] += 1
+            elif vote.vote == -1:
+                votes["down"].append(vote.user.pk)
+                votes["down_count"] += 1
+            votes["count"] = votes["up_count"] - votes["down_count"]
+            votes["point"] = votes["count"]
```

Comment:
Update the point and count values.
votes["point"] = votes["up_count"] - votes["down_count"]
votes["count"] = votes["count"]

Reply:
Done